### PR TITLE
LPD-6870 Sign In Utility Page

### PR DESCRIPTION
**Pull request by oliverkecskemety**
[LPD-6870](https://liferay.atlassian.net/browse/LPD-6870)

### CODE REVIEW

**Comment by oliverkecskemety** [comment]
ci:test:sf

**Comment by oliverkecskemety** [comment]
ci:test:sf

**Comment by oliverkecskemety** [comment]
ci:test:sf

**Comment by oliverkecskemety** [comment]
ci:test:sf

**Review by stian-sigvartsen — COMMENTED** [review]
This is good progress. I did not expect that we would be able to serve the content at a static URL like that in a reliable way, but this looks very close to be achieving that.

In testing it I had some issues.
1) After creating a custom site and adding a default Sign In utility page there, it kept rendering the default site's utility page instead. When debugging it was getting that site from ThemeDisplay#getScopeGroupId.
2) As we discussed offline, portlet URLs created by the `LoginPortlet` point to a different page. Presumably we just need a bit more ThemeDisplay initialization

**Comment by stian-sigvartsen** [comment]
As we discussed today offline, we can defer [2] to another follow-up ticket. I think it would be more important to deal with [1] in this current ticket.

File: modules/apps/layout/layout-utility-page/layout-utility-page-sign-in/src/main/java/com/liferay/layout/utility/page/sign/in/internal/struts/SignInStrutsAction.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import org.jsoup.nodes.Element;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Olivér Kecskeméty
+ */
+@Component(property = "path=/portal/sign_in", service = StrutsAction.class)
+public class SignInStrutsAction implements StrutsAction {
```
Comment (stian-sigvartsen) [inline modules/apps/layout/layout-utility-page/layout-utility-page-sign-in/src/main/java/com/liferay/layout/utility/page/sign/in/internal/struts/SignInStrutsAction.java:34]: Did you consider adding the logic from this Struts Action into the existing `LoginAction`?
Then there would not be a new DXP path to support. Just curious about the reasoning.

**Comment by oliverkecskemety** [comment]
ci:test:sf

**Comment by oliverkecskemety** [comment]
ci:test:sf

File: modules/apps/layout/layout-utility-page/layout-utility-page-sign-in/src/main/java/com/liferay/layout/utility/page/sign/in/internal/struts/SignInStrutsAction.java (hunk 1 above)
Comment (oliverkecskemety) [reply modules/apps/layout/layout-utility-page/layout-utility-page-sign-in/src/main/java/com/liferay/layout/utility/page/sign/in/internal/struts/SignInStrutsAction.java:34]: Yes, and István also raised this question. I decided to follow the pattern as the 500 status and the Terms of Use utility pages also defined struts actions in their modules.

**Comment by ling-alan-huang** [comment]
This [PR](https://github.com/brianchandotcom/liferay-portal/pull/147646) can fix SF issue, Brain hasn't published to master yet.

**Comment by stian-sigvartsen** [comment]
FYI

Some of the changes in this PR have been sent to BChan for merge on brianchandotcom/liferay-portal/pull/147744 because there was a technical dependency.

So we will need to rebase this ticket if that is merged.

**Comment by stian-sigvartsen** [comment]
I had some reservations about introducing a 2nd Struts Action which from a client perspective has seemingly the same purpose as `LoginAction`.

After exploring the problem deeper I found it is necessary only because we need to inject the LoginPortlet namespace into the redirect parameter so it can be read by the LoginPortlet on the UP.

We are essentially providing a tactical solution to the lack of a friendly URL for utility page instances. When that is solved, we can remove the 2nd Struts action.

I simplified parts of the solution, please see liferay-appsec/liferay-portal/pull/1671

**Pull request closed without merging**